includes/icon-functions.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * SVG icons related functions
+ *
+ * @package TenUpScaffold
+ */
+
+/**
+ * An SVG template helper with robust array of options.
+ *
+ * Example:
+ * tenupscaffold_svg_icon(
+ *     array(
+ *         'desc'   => 'Follow 10up on Instagram',
+ *         'fill'   => '#f4428f',
+ *         'height' => 24,
+ *         'icon'   => 'instagram',
+ *         'output' => true,
+ *         'role'   => 'img',
+ *         'title'  => 'Follow 10up on Instagram',
+ *         'width'  => 24,
+ *     ),
+ * );
+ *
+ * @link   https://10up.github.io/Engineering-Best-Practices/markup/#svg-embedded-in-html
+ * @param  array $args   The SVG options such as Aria role, dimensions, title.
+ * @return html
+ */
+function tenupscaffold_svg_icon( $args = array() ) {
+	// Make sure $args are an array.
+	if ( empty( $args ) ) {
+		return esc_html__( 'Please define default parameters in the form of an array.', 'tenupscaffold' );
+	}
+
+	// Define an icon.
+	if ( false === array_key_exists( 'icon', $args ) ) {
+		return esc_html__( 'Please define an SVG icon name.', 'tenupscaffold' );
+	}
+
+	// Set defaults.
+	$defaults = array(
+		'desc'   => '',
+		'fill'   => '',
+		'height' => '',
+		'icon'   => '',
+		'output' => false,
+		'role'   => 'presentation',
+		'title'  => '',
+		'width'  => '',
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Figure out which title to use.
+	$block_title = ( $args['title'] ) ? $args['title'] : $args['icon'];
+
+	// Generate random IDs for the title and description.
+	$random_number  = wp_rand( 0, 99999 );
+	$block_title_id = 'title-' . sanitize_title( $block_title ) . '-' . $random_number;
+	$desc_id        = 'desc-' . sanitize_title( $block_title ) . '-' . $random_number;
+
+	// Set ARIA.
+	$aria_hidden     = ' aria-hidden="true"';
+	$aria_labelledby = '';
+	$aria_role       = ( $args['role'] ) ? ' role="' . $args['role'] . '"' : '';
+
+	// If we have a title and description then let's assign appropriate Aria.
+	if ( $args['title'] && $args['desc'] ) {
+		$aria_labelledby = ' aria-labelledby="' . $block_title_id . ' ' . $desc_id . '"';
+		$aria_hidden     = '';
+	}
+
+	// Set SVG parameters.
+	$fill   = ( $args['fill'] ) ? ' fill="' . $args['fill'] . '"' : '';
+	$height = ( $args['height'] ) ? ' height="' . $args['height'] . '"' : '';
+	$width  = ( $args['width'] ) ? ' width="' . $args['width'] . '"' : '';
+
+	$allowed_tags = tenupscaffold_get_post_with_svg_allowed_tags();
+
+	// Start a buffer...
+	ob_start();
+	?>
+
+	<svg
+	<?php
+		echo $fill; // WPCS XSS OK.
```

Comment:
All of these items need to be escaped using `esc_attr()`. You could escape them above but it would be better to late escape them here, especially as this may be used on a VIP project.

Example:
```
$fill = ( $args['fill'] ) ? ' fill="' . $args['fill'] . '"' : false;

<?php if ( $fill ) : ?>
fill="<?php esc_attr( $fill  ); ?>
<?php endif ;?>

Reply:
@ryansommers I updated the logic to check for value and escaped late with `est_attr()`. I feel like the logic could be a little DRYer, but meh, it works. 

Thanks for the extra eyes and feedback!

Reply:
That @ryansommers is ALWAYS stealing my karm and comments :)

Reply:
@ryanwelcher as it should be.